controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -251,29 +258,38 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 	switch {
 	// We are creating the first replica
 	case numMachines < desiredReplicas && numMachines == 0:
-		// create new Machine w/ init
-		logger.Info("Scaling to 1", "Desired Replicas", desiredReplicas, "Existing Replicas", numMachines)
+		// Create new Machine w/ init
+		logger.Info("Initializing control plane", "Desired", desiredReplicas, "Existing", numMachines)
 		if err := r.initializeControlPlane(ctx, cluster, kcp); err != nil {
-			logger.Error(err, "Failed to initialize the Control Plane")
-			r.recorder.Eventf(kcp, corev1.EventTypeWarning, "FailedInitialization", "Failed to initialize the control plane: %v", err)
+			logger.Error(err, "Failed to initialize control plane")
+			r.recorder.Eventf(kcp, corev1.EventTypeWarning, "FailedInitialization", "Failed to initialize cluster %s/%s control plane: %v", cluster.Namespace, cluster.Name, err)
 			return ctrl.Result{}, err
```

Comment:
This comment applies to all three cases, initialize, scale up, scale down:

We don't want to retry the health checks immediately. Eventually controller-runtime gives up and does not retry when an error is returned. This means that after 10 failed health checks (which will happen fairly quickly) we have to wait for a full re-reconcile to try again. This is bad because that's a 10 minute pause when trying to scale up.

Because of this, all three cases should instead return `return ctrl.Result{RequeueAfter: 5 * time.Second}, nil` (or whatever duration makes sense, i think even a 10-20 second duration would be appropriate here.

Reply:
Thanks a lot for explaining this.

I think I observed that behavior in my CAPD cluster yesterday. but there was nothing in the log. Would it help to log when the controller is waiting for a full re-reconcile?

Reply:
this wouldn't be possible from the controller, we'd have to change controller-runtime to do this. But generally, the reconciler is always in a state of waiting to do something. A reconcile loop is fairly quick so you can just kind of assume it's always waiting for a reconcile. It's impossible to predect when it's going to reconcile next because in the time it's waiting for a re-reconcile due to the 10 minute time out, some object could change triggering a reconcile anyway.

Reply:
Done.

```
	// HealthCheckFailedRequeueAfter is how long to wait before trying to scale
	// up/down if some target cluster health check has failed
	HealthCheckFailedRequeueAfter = 20 * time.Second
```